### PR TITLE
OSDOCS#11457: User-defined tags changes

### PR DESCRIPTION
**Pull request by snarayan-redhat**
Version(s): 4.17

Issue: https://issues.redhat.com/browse/OSDOCS-11457

Link to docs preview: https://79964--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure/ipi/installing-azure-customizations.html#installing-azure-user-defined-tags_installing-azure-customizations

Review:
- [x] QE has approved this change. @jinyunma

### CODE REVIEW

**Comment by jinyunma** [comment]
LGTM

File: modules/installation-user-defined-tags-requirements-azure.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+The user-defined tags have the following requirements:
+
+* A tag key can have a maximum of 128 characters.
+
+* A tag key must begin with a letter, end with a letter, number or underscore, and can contain only letters, numbers, underscores(`_`), periods(`.`), and hyphens(`-`).
+
+* Tag keys are case-insensitive.
+
+* Tag keys cannot be `name`. It cannot have prefixes such as `kubernetes.io`, `openshift.io`, `microsoft`, `azure`, and `windows`.
```
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-requirements-azure.adoc:15]:
```suggestion
* A tag key must not be called `name`. 

* A tag key must not use the following prefixes:

** `azure`
** `kubernetes.io`
** `microsoft`
** `openshift.io`
** `windows`
```

File: modules/installation-user-defined-tags-requirements-azure.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// * installing/installing_azure/installing-azure-customizations.adoc
+:_mod-docs-content-type: REFERENCE
+[id="installation-user-defined-tags-requirements-azure_{context}"]
+= User-defined tags requirements
+
+The user-defined tags have the following requirements:
+
+* A tag key can have a maximum of 128 characters.
+
+* A tag key must begin with a letter, end with a letter, number or underscore, and can contain only letters, numbers, underscores(`_`), periods(`.`), and hyphens(`-`).
```
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-requirements-azure.adoc:11]:
```suggestion
* A tag key must begin with a letter.

* A tag key must end with a letter, number or underscore (`_`).

* A tag key must contain only letters, numbers, underscores (`_`), periods (`.`), or hyphens (`-`).
```

File: modules/installation-user-defined-tags-requirements-azure.adoc
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-requirements-azure.adoc:1]: Some of these requirements are a little easier to follow if broken out of long sentences, and they should all have parallel construction to the extent possible

File: modules/installation-user-defined-tags-azure.adoc [hunk 3]
```diff
@@ -3,78 +3,16 @@
 
 :_mod-docs-content-type: CONCEPT
 [id="installing-azure-user-defined-tags_{context}"]
-= Configuring user-defined tags for Azure
+= Configuring user-defined tags for {azure-short}
 
-In {product-title}, you can use the tags for grouping resources and for managing resource access and cost. You can define the tags on the Azure resources in the `install-config.yaml` file only during {product-title} cluster creation. You cannot modify the user-defined tags after cluster creation.
+In {product-title}, you can use tags for grouping resources and for managing resource access and cost. Tags are applied only to the resources created by the {product-title} installation program and its core operators such as Machine API Operator, Cluster Ingress Operator, Cluster Image Registry Operator. The {product-title} consists of the following types of tags:
```
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-azure.adoc:8]:
```suggestion
In {product-title}, you can use tags for grouping resources and for managing resource access and cost. Tags are applied only to the resources created by the {product-title} installation program and its core Operators such as Machine API Operator, Cluster Ingress Operator, Cluster Image Registry Operator. The {product-title} consists of the following types of tags:
```

Comment (jeana-redhat) [inline modules/installation-user-defined-tags-azure.adoc:8]: I am not sure what this means:

`The {product-title} consists of the following types of tags:`

> The OpenShift Container Platform consists of the following types of tags:

"OpenShift Container Platform uses the following types of tags" maybe?

File: modules/installation-user-defined-tags-azure.adoc [hunk 4]
```diff
@@ -3,78 +3,16 @@
...
 
-User-defined and {product-title} specific tags are applied only to the resources created by the {product-title} installer and its core operators such as Machine api provider azure Operator, Cluster Ingress Operator, Cluster Image Registry Operator.
+* User-defined tags: User-defined tags are manually created in `install-config.yaml` file during installation. When creating the user-defined tags, you must consider the following points:
 
-By default, {product-title} installer attaches the {product-title} tags to the Azure resources. These {product-title} tags are not accessible for the users.
+** User-defined tags on {azure-short} resources can only be defined during {product-title} cluster creation, and cannot be modified after the cluster is created.
 
-You can use the `.platform.azure.userTags` field in the `install-config.yaml` file to define the list of user-defined tags as shown in the following `install-config.yaml` file.
+** Support for user-defined tags is available only for the resources created in the {azure-short} Public Cloud.
```
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-azure.adoc:16]: Consider using a [description list](https://docs.asciidoctor.org/asciidoc/latest/lists/description/) for items like this, i.e.:

```
{product-title} tags:: By default, {product-title} installer attaches the {product-title} tags to the {azure-short} resources. These {product-title} tags are not accessible for the users. 
+
The format of the {product-title} tags is `kubernetes.io_cluster.<cluster_id>:owned` where `cluster_id` parameter is the value of `.status.infrastructureName` in `config.openshift.io/Infrastructure` API.

User-defined tags:: User-defined tags are manually created in `install-config.yaml` file during installation. When creating the user-defined tags, you must consider the following points:
+
* User-defined tags on {azure-short} resources can only be defined during {product-title} cluster creation, and cannot be modified after the cluster is created.
* Support for user-defined tags is available only for the resources created in the {azure-short} Public Cloud.
* User-defined tags are not supported for the {product-title} clusters upgraded to {product-title} {product-version}.
```

That looks something like this (VSC preview doesn't render the attributes, but so you can see the format):
image

File: modules/installation-user-defined-tags-azure.adoc [hunk 5]
```diff
@@ -3,78 +3,16 @@
...
+In {product-title}, you can use tags for grouping resources and for managing resource access and cost. Tags are applied only to the resources created by the {product-title} installation program and its core operators such as Machine API Operator, Cluster Ingress Operator, Cluster Image Registry Operator. The {product-title} consists of the following types of tags:
 
-Support for user-defined tags is available only for the resources created in the Azure Public Cloud. User-defined tags are not supported for the {product-title} clusters upgraded to {product-title} {product-version}.
+* {product-title} tags: By default, {product-title} installer attaches the {product-title} tags to the {azure-short} resources. These {product-title} tags are not accessible for the users. The format of the {product-title} tags is `kubernetes.io_cluster.<cluster_id>:owned` where `cluster_id` parameter is the value of `.status.infrastructureName` in `config.openshift.io/Infrastructure` API.
```
Comment (jeana-redhat) [inline modules/installation-user-defined-tags-azure.adoc:10]:
```suggestion
* {product-title} tags: By default, {product-title} installer attaches the {product-title} tags to the {azure-short} resources. These {product-title} tags are not accessible to users. The format of the {product-title} tags is `kubernetes.io_cluster.<cluster_id>:owned`, where `<cluster_id>` is the value of `.status.infrastructureName` in the infrastructure resource for the cluster.
```

**Review by jeana-redhat — COMMENTED** [review]
Nice work breaking this out! Left a few things to consider. :nerd_face: 

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/installation-user-defined-tags-azure.adoc (hunk 4 above)
Comment (snarayan-redhat) [reply modules/installation-user-defined-tags-azure.adoc:16]: sure, thanks!

File: modules/installation-user-defined-tags-azure.adoc (hunk 5 above)
Comment (snarayan-redhat) [reply modules/installation-user-defined-tags-azure.adoc:10]: Thanks Jeana! I re-read and I have couple more fixes now :D

```suggestion
* {product-title} tags: By default, {product-title} installation program attaches the {product-title} tags to the {azure-short} resources. These {product-title} tags are not accessible to the users. The format of the {product-title} tags is `kubernetes.io_cluster.<cluster_id>:owned`, where `<cluster_id>` is the value of `.status.infrastructureName` in the infrastructure resource for the cluster.
```

File: modules/installation-user-defined-tags-azure.adoc (hunk 3 above)
Comment (snarayan-redhat) [reply modules/installation-user-defined-tags-azure.adoc:8]: It means "OCP can have the following types of tags".

File: modules/installation-user-defined-tags-requirements-azure.adoc (hunk 1 above)
Comment (snarayan-redhat) [reply modules/installation-user-defined-tags-requirements-azure.adoc:15]: Thanks a lot for pointing this out! I think I totally missed reviewing this piece after pulling it out in to this module.

**Comment by jeana-redhat** [comment]
Looking great :grinning: :tada:

**Pull request merged**